qt/widgets/instrumentview/src/InstrumentWidget.cpp

```diff
@@ -763,6 +763,10 @@ void InstrumentWidget::updateIntegrationWidget(bool init) {
     } else {
       m_xIntegration->setRange(0, 0);
     }
+  } else {
```

Comment:
I'm just looking at the comment in the `init` `if` above. Have we tried replacing workspace here? The comment above suggests there may be an issue if `setRange` (called by `setWholeRange`) is called if `init` is True?

Reply:
Yes, the workspace is replaced after loading data for the first time. The first time `updateIntegrationWidget` is called `minRange` and `maxRange` are 0. After loading data successfully `updateIntegrationWidget` is called again (with `init` set to True) with the new values for `minRange` and `maxRange`. The instrument viewer has already been initialized by that time.